iclaims.go

```diff
@@ -109,3 +101,172 @@ func ValidateClaims(c IClaims) error {
+// stated profile.
+func DecodeClaimsFromCBOR(buf []byte) (IClaims, error) {
+	selector := struct {
+		// note: code point 265 is defined as the eat_profile claim in
+		// EAT(https://datatracker.ietf.org/doc/draft-ietf-rats-eat/).
+		// This is not specific to PSA, and so is not something that we
+		// expect PSA profiles to be able to override.
+		// The only exception to this P1, which does not use the
+		// eat_profile claim (code point 265). It instead defines its
+		// own pofile at code point -7500. This will not be unmarshaled
+		// into this field. For P1, the dm.Unmarshal below will leave
+		// the field in its default value "" (however, it not result in
+		// an error). As the P1 profile field is optional, if a profile
+		// is not present in the claims, they are assumed to be for P1
+		// (as that would make them automatically invalid for any other
+		// profile). Because of this, the P1 claims will be selected
+		// from the register if the Profile field here is empty, and
+		// the decoding will proceed correctly. P1's -7500 profile
+		// field will then be validated as part of the full claims
+		// decoding in FromUnvalidatedCBOR() further down.
+		Profile string `cbor:"265,keyasint"`
+	}{}
+
+	err := dm.Unmarshal(buf, &selector)
+	if err != nil {
+		return nil, err
+	}
+
+	entry, ok := profilesRegister[selector.Profile]
+	if !ok {
+		return nil, fmt.Errorf("unknown profile: %q", selector.Profile)
+	}
+
+	claims := entry.Profile.GetClaims()
+
+	if err := dm.Unmarshal(buf, claims); err != nil {
+		return nil, err
+	}
+
+	return claims, nil
+}
+
+// Deprecated: use DecodeAndValidateClaimsFromJSON instead.
+func DecodeJSONClaims(buf []byte) (IClaims, error) {
+	return DecodeAndValidateClaimsFromJSON(buf)
+}
+
+// DecodeAndValidateClaimsFromJSON returns an IClaims implementation instance populated
+// from the provided JSON buf. The implementation used is determined by value
+// of the profile field in the provided JSON object. In the absence of
+// this field, Profile1 (PSA_IOT_PROFILE_1) is assumed. Claims are validated
+// according to the profile as part of decoding.
+func DecodeAndValidateClaimsFromJSON(buf []byte) (IClaims, error) {
+	claims, err := DecodeClaimsFromJSON(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := claims.Validate(); err != nil {
+		return nil, err
+	}
+
+	return claims, nil
+}
+
+// Deprecated: use DecodeClaimsFromJSON instead.
+func DecodeUnvalidatedJSONClaims(buf []byte) (IClaims, error) {
+	return DecodeClaimsFromJSON(buf)
+}
+
+// DecodeClaimsFromJSON returns an IClaims implementation instance
+// populated from the provided JSON buf. The implementation used is determined
+// by value of the profile field in the provided JSON object. In the absence
+// of this field, Profile1 (PSA_IOT_PROFILE_1) is assumed. No validation is
+// performed to confirm that the decoded claims actually conform to the stated
+// profile.
+func DecodeClaimsFromJSON(buf []byte) (IClaims, error) {
```

Comment:
I think, we ave been using simple `FromJSON` and `ToJSON` naming for most of the APIs doing this purpose!

Silimary FromCBOR and ToCBOR

Reply:
For methods, yes. And  now ther ere equivalent `UnmarshalJSON`, etc.

For functions (that do not have context of existing object), we've already been using `DecodeCalaimsXX` (note we need to say "claims" in the function because there is an equivalent for evidence).

Reply:
ok, i agree, we can resolve this conversation. I have already approved it!